Character/Stats.cpp

```diff
@@ -65,6 +65,7 @@ void Stats::add(const Stats* rhs) {
     increase_ranged_hit(rhs->get_ranged_hit_chance());
     increase_ranged_crit(rhs->get_ranged_crit_chance());
     increase_attack_speed(rhs->get_attack_speed());
+    increase_ranged_attack_speed(rhs->get_ranged_attack_speed_percent());
```

Comment:
I was a bit surprised to see there is no decrease_ranged_attack_speed in Stats::remove(), but I've happened upon (and I think you might have as well) a deeper issue in Stats element when it comes to both melee and ranged attack speed. I will note some thoughts here, but I consider them out of scope for this pull request.

Stats is intended to not be a "compound stat" element where e.g. multiple sources of bonuses are included, that is instead the intended role of the CharacterStats class. A Stats instance is/was supposed to be a single source of values, and so supporting the addition of multiple attack speed increases is a bit strange in that sense (you'd never see e.g. a single item with multiple, multiplicative attack speed bonuses for example).

I see there are remnants of unused melee attack speed functions and variables which hints toward the intended use, but actually using them is not currently possible since any change to a Stats::increase_melee_attack_speed actually does nothing, you'll have to invoke CharacterStats::increase... or Character::increase_.. . Note: the difference between the two is whether or not the user is in combat, so for example temp buffs like Aspect of the  Hawk would use Character::, and equip bonuses would use CharacterStats::. A bit confusing, which is not optimal.

Reply:
I have to say that I was a bit confused at the beginning about the Stats class as it didn't have any impact on the attack speed. What do you want me to do here? Delete this line?

Reply:
I'm fine with leaving it as is in this pull request, I'm thinking I'll refactor the Stats class a bit this weekend.